nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ReplaceText.java

```diff
@@ -586,52 +672,76 @@ public void process(final InputStream in, final OutputStream out) throws IOExcep
                         try (final LineDemarcator demarcator = new LineDemarcator(in, charset, maxBufferSize, 8192);
                             final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(out, charset))) {
 
-                            String oneLine;
 
-                            final StringBuffer sb = new StringBuffer();
+//                            final StringBuffer sb = new StringBuffer();
                             Matcher matcher = null;
-
-                            while (null != (oneLine = demarcator.nextLine())) {
-                                additionalAttrs.clear();
-                                if (matcher == null) {
-                                    matcher = searchPattern.matcher(oneLine);
+                            String precedingLine = demarcator.nextLine();
```

Comment:
These lines of code can be seen in each `ReplacementStrategyExecutor` implementations and it makes harder to maintain. Please try consolidating them by making this logic around Evaluation modes to a shared common method.

For example, by extracting these lines to a method and pass the each ReplacementStrategyExecutor specific part as a function.

Reply:
I tried refactoring common code to the best of my knowledge. I would welcome suggestions to improve it.